File: backend/src/models/Deity.js
import mongoose from "mongoose";

const deitySchema = new mongoose.Schema({
  name: {
    en: { type: String, required: true },
    ml: { type: String },
    sa: { type: String }
  },
  slug: { type: String, required: true, unique: true },
  shortDescription: { type: String, required: true },
  fullDescription: { type: String, required: true },
  pronunciationGuide: String,
  tradition: { type: String, required: true },
  iconUrl: String,
  imageUrl: String,
  isFeatured: { type: Boolean, default: false },
  order: { type: Number, default: 0 }
});

export const Deity = mongoose.model("Deity", deitySchema);

File: shared/src/commonMain/kotlin/com/divya/data/models/Deity.kt
package com.divya.data.models

import kotlinx.serialization.Serializable

@Serializable
data class Deity(
    val id: String,
    val slug: String,
    val name: LocalizedText,
    val shortDescription: String,
    val fullDescription: String,
    val pronunciationGuide: String? = null,
    val tradition: String,
    val iconUrl: String? = null,
    val imageUrl: String? = null,
    val isFeatured: Boolean = false,
    val order: Int = 0,
)

